Lib/urllib/request.py

```diff
@@ -1136,15 +1136,15 @@ def get_authorization(self, req, chal):
         A2 = "%s:%s" % (req.get_method(),
                         # XXX selector: what about proxies and full urls
                         req.selector)
-        if qop == 'auth':
+        if 'auth' in qop.split(','):
             if nonce == self.last_nonce:
                 self.nonce_count += 1
             else:
                 self.nonce_count = 1
                 self.last_nonce = nonce
             ncvalue = '%08x' % self.nonce_count
             cnonce = self.get_cnonce(nonce)
-            noncebit = "%s:%s:%s:%s:%s" % (nonce, ncvalue, cnonce, qop, H(A2))
+            noncebit = "%s:%s:%s:%s:%s" % (nonce, ncvalue, cnonce, 'auth', H(A2))
```

Comment:
* Don't we need to account for `auth-int` in the noncebit?
* We will also need a test to go along with this change.

Reply:
As far as I could tell, using 'auth,auth-int' as found in the server capabilities led to a wrong nonce that got rejected by the server. It makes sense since we have to select either 'auth' or 'auth-int' when sending our renewed connection request. §3.2.2.1 of RFC 2617 suggests that the value used to compute noncebit should either be 'auth' or 'auth-int'.

server challenge:
`WWW-Authenticate: Digest realm="onvif", qop="auth,auth-int", nonce="5dcd4c06ca893ade7254", opaque="528eacb9"`

reply with `..., cnonce, 'auth'`, accepted by the server
`Authorization: Digest username="admin", realm="onvif", nonce="5dcd4c06ca893ade7254", uri="/onvif/device_service", response="7bf0da0d069068282a7e84e6a782d555", opaque="528eacb9", algorithm="MD5", qop=auth, nc=00000001, cnonce="69c742af8d731cd5"

reply with `..., cnonce, 'auth,auth-int`', ...` still has qop=auth in `Authorization:` and is rejected by the server.

Or did I mis-interpreted your comment ?

Reply:
As for adding test case, do you have another (HTTP-Digest-related) test case you would suggest to use as a base ?

Reply:
Hi @PypeBros  

There are specific guidelines in https://tools.ietf.org/html/rfc2617 on how to behave when "qop" directive's value is "auth" or "auth-int".

Specifically, there is a change in A2 when a qop is auth-int

```
3.2.2.3 A2

   If the "qop" directive's value is "auth" or is unspecified, then A2
   is:

      A2       = Method ":" digest-uri-value

   If the "qop" value is "auth-int", then A2 is:

      A2       = Method ":" digest-uri-value ":" H(entity-body)
```

The same is explained in the Wikipedia article too: https://en.wikipedia.org/wiki/Digest_access_authentication

So, I am not sure if this change accomplishes the support for 'auth-int' properly if the server responds with that in qop.

It seems like we have to calculate `respdig` (line 1148 and line 1150) properly depending on if the qop is either `auth` or `auth-int`
---

By test case, I was assuming if `test_basic_and_digest_auth_handlers` could be expanded, but it looks like we might have to come up a more involved test case here as I could not find some that could be easily expanded upoin the stdlib.

Reply:
the purpose of this change is not to add `auth-int` support. Only to fix existing `auth` support when communicating with a server that supports both auth and auth-int modes.

Reply:
@PypeBros: Got it. I think this is reasonable. We could add a comment to this effect above the line `if 'auth' in qop.split(','):` - which explains clearly that we 

```
As per  RFC 2617, when server sends auth,auth-int, the client could use either `auth` or `auth-int` to the response back. we use `auth` to send the response back. 
TODO: Support for `auth-int` is not supported yet.
```

With this addition, this PR is good to merge.